test/admin/functions.test.js

```diff
@@ -0,0 +1,75 @@
+const StitchMongoFixture = require('../fixtures/stitch_mongo_fixture');
+
+import { getAuthenticatedClient } from '../testutil';
+
+describe('Functions V3', () => {
+  let test = new StitchMongoFixture();
+  let functions;
+  let app;
+  let apps;
+  beforeAll(() => test.setup());
+  afterAll(() => test.teardown());
+  beforeEach(async () => {
+    let adminClient = await getAuthenticatedClient(test.userData.apiKey.key);
+    test.groupId = test.userData.group.groupId;
+    apps = await adminClient.v2().apps(test.groupId);
+    app = await apps.create({ name: 'testname' });
+    functions = adminClient
+      .v2()
+      .apps(test.groupId)
+      .app(app._id)
+      .functions();
+  });
+  afterEach(async () => {
+    await apps.app(app._id).remove();
+  });
+
+  it('listing functions should return an empty list', async () => {
+    let funcs = await functions.list();
+    expect(funcs).toHaveLength(0);
+  });
+
+  it('creating functions should work', async () => {
+    let func = await functions.create({ name: 'myFunction' });
+    expect(func.name).toEqual('myFunction');
+
+    let funcs = await functions.list();
+    expect(funcs).toHaveLength(1);
+    expect(funcs[0].name).toEqual('myFunction');
+  });
+
+  it('invalid create requests should fail', async () => {
+    await expect(functions.create({ name: '' })).rejects.toBeDefined();
+  });
+
+  it('fetching function should work', async () => {
+    let newFunc = await functions.create({ name: 'myFunction' });
+    expect(newFunc._id).toBeTruthy();
+
+    let func = await functions.function(newFunc._id).get();
+    expect(func.name).toEqual('myFunction');
+  });
+
+  it('deleting function should work', async () => {
+    let func = await functions.create({ name: 'myFunction' });
+    expect(func._id).toBeTruthy();
+
+    let funcs = await functions.list();
+    expect(funcs).toHaveLength(1);
+
+    await functions.function(func._id).remove();
+
+    funcs = await functions.list();
+    expect(funcs).toHaveLength(0);
+  });
+
+  it('updating function should work', async () => {
+    let func = await functions.create({ name: 'myFunction' });
+    expect(func._id).toBeTruthy();
+
+    await functions.function(func._id).update({ name: 'myNewFunction' });
```

Comment:
can we also include a change to the `source` field of the function as well (with a corresponding assertion after the update)?